mmdet/models/utils/position_encoding.py

```diff
@@ -0,0 +1,149 @@
+import math
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import uniform_init
+
+from .builder import POSITION_ENCODING
+
+
+@POSITION_ENCODING.register_module()
+class SinePositionEmbedding(nn.Module):
+    """Position encoding with sine and cosine functions.
+
+    See `End-to-End Object Detection with Transformers
+    <https://arxiv.org/pdf/2005.12872>`_ for details.
+
+    Args:
+        num_feats (int): The feature dimension for each position
+            along x-axis or y-axis. Note the final returned dimension
+            for each position is 2 times of this value.
+        temperature (int): The temperature used for scaling the position
+            embedding. Default 10000.
+        normalize (bool): Whether to normalize the position embedding.
```

Comment:
If the argument has a default value, optional should be added in the type, like `(bool, optional)`, `(float, optional)`. Please check other places in the code and modify it accordingly.